src/test/java/org/mule/test/http/functional/HttpListenerOverloadTestCase.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.http.functional;
+
+import static java.lang.String.format;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mule.functional.api.component.FunctionalTestProcessor.getFromFlow;
+import static org.mule.runtime.http.api.HttpConstants.HttpStatus.OK;
+import static org.mule.runtime.http.api.HttpConstants.HttpStatus.SERVICE_UNAVAILABLE;
+import static org.mule.test.http.AllureConstants.HttpFeature.HTTP_EXTENSION;
+
+import org.mule.functional.api.component.FunctionalTestProcessor;
+import org.mule.runtime.api.util.concurrent.Latch;
+import org.mule.runtime.core.api.util.IOUtils;
+import org.mule.tck.junit4.rule.DynamicPort;
+
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import io.qameta.allure.Feature;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.fluent.Executor;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+@Feature(HTTP_EXTENSION)
```

Comment:
No need for this since it's in the super class, add a `Story` annotation though so we can track this is for the overload scenario.